src/sql/workbench/browser/designer/designer.ts

```diff
@@ -346,12 +370,41 @@ export class Designer extends Disposable implements IThemable {
 			}
 			if (edit.type === DesignerEditType.Add) {
 				// For tables in the main view, move focus to the first cell of the newly added row, and the properties pane will be showing the new object.
-				if (edit.path.length === 1) {
+				if (edit.path.length === 2) {
+					const propertyName = edit.path[0] as string;
+					const index = edit.path[1] as number;
+					const table = this._componentMap.get(propertyName).component as Table<Slick.SlickData>;
+					const tableProperties = this._componentMap.get(propertyName).defintion.componentProperties as DesignerTableProperties;
+					let selectedCellIndex = tableProperties.itemProperties.findIndex(p => p.componentType === 'input');
```

Comment:
why input only?

Reply:
the changes in this function seem to be more than what's needed. I think all that we need is:
1. get row index
2. get cell index (1 if canMoveRows otherwise 0)
3. table.setActiveCell(row, cell);

especially the code to rerender the grid is not needed.


similar to the move edit post handler.

Reply:
also, there seems to be a need to extract out the logic to reuse the 
```
					try {
						table.setActiveCell(index, selectedCellIndex);
					}
					catch {
						// Ignore the slick grid error when setting active cell.
					}
```